Zend/tests/lazy_objects/gh18038-003.phpt

```diff
@@ -0,0 +1,30 @@
+--TEST--
+GH-18038 003: Lazy proxy calls magic methods twice
+--FILE--
+<?php
+
+#[AllowDynamicProperties]
+class C {
+    public $_;
+    public function __get($name) {
+        var_dump(__METHOD__);
+        return $this->$name;
+    }
+}
+
+$rc = new ReflectionClass(C::class);
+
+$obj = $rc->newLazyProxy(function () {
+    echo "init\n";
+    return new C;
+});
+
+var_dump($obj->prop);
+
+?>
+--EXPECTF--
```

Comment:
I don't understand why the `__get` function should get called before instantiating the object? I.e. why does "C::__get" appear before "init"?

Reply:
It’s how lazy objects are supposed to behave: we interact primarily with the proxy (we execute the proxy’s code), but any access to its state is forwarded to the real instance.

So here we execute the getter, which tries to access a property. This triggers initialization before the access is forwarded.

Reply:
Okay I see, not what I intuitively expected but it makes sense

Reply:
I suppose there's also an implicit assumption here that the initialized properties does not contain dynamic properties. If it does, the initializer would need to be called before `__get`, to ensure that the property is actually still does not exist after the initializer has been called.